### ISSUE TITLE
`Transfer-Encoding: chunked` response support

### ISSUE THREAD

**Issue by huntharo** [H4ad/serverless-adapter#165]
## Feature Request

Remove chunked encoding on responses instead of letting the encoding go through and have a `content-length` be calculated including the encoding.

**Is your feature request related to a problem? Please describe.**

When `curl`'d via an AWS ALB, the route below returns the result below, which includes the chunked response encoding control bytes in the response payload.

```
< HTTP/2 200 
< server: awselb/2.0
< date: Mon, 01 Jan 2024 05:58:45 GMT
< content-type: text/plain
< content-length: 65
< x-powered-by: Express
< 
19
INITIAL PAYLOAD RESPONSE

17
FINAL PAYLOAD RESPONSE

0
```

Express Route:

```
app.get("/chunked-response", async (req, res) => {
  // Send headers right away
  res.setHeader("Content-Type", "text/plain");
  res.setHeader("Transfer-Encoding", "chunked");
  res.status(200);

  // Send initial payload
  res.write("INITIAL PAYLOAD RESPONSE\n");

  // Wait for 5 seconds
  await sleep(5000);

  // Send final payload and close out the response
  res.end("FINAL PAYLOAD RESPONSE\n");
});
```

**Describe the solution you'd like**

I don't believe an ALB invoking a Lambda supports `Transfer-Encoding: chunked` at all, but there are cases where code will exist that writes the response as chunked because it doesn't know the content length ahead of time.  Yes, in that case the entire response is going to be buffered within the Lambda so the length will end up being known.

It would be beneficial if the `Transfer-Encoding: chunked` header was stripped and the control characters were removed before the response body was returned, OR, at the very least throw a nasty exception if a `Transfer-Encoding: chunked` response header is encountered.

**Describe alternatives you've considered**

Not using `Transfer-Encoding: chunked`... however, the specific case that I'm handling is a comparison test of when the first bytes to a response arrive from this invocation method vs via another method that I'm working on, so I have to use chunked if I want the Express code to be compariable.

## Are you willing to resolve this issue by submitting a Pull Request?

Potentially.

- [ ] Yes, I have the time, and I know how to start.
- [x] Yes, I have the time, but I don't know how to start. I would need guidance.
- [ ] No, I don't have the time, although I believe I could do it if I had the time...
- [ ] No, I don't have the time and I wouldn't even know how to start.

**Comment by H4ad** [comment]
From what I saw in other libraries, the issue was fixed by manually removing those control characters, ex:
https://github.com/dougmoscrop/serverless-http/commit/e41bd784e3351d18b7c3f8aec4fcead764f396af

If you want to try add support for this feature, you can start modifying the following code:

https://github.com/H4ad/serverless-adapter/blob/main/src/adapters/aws/alb.adapter.ts#L133-L155

To be something like this:

```ts
/**
   * {@inheritDoc}
   */
  public getResponse({
    event,
    headers: responseHeaders,
    body,
    isBase64Encoded,
    statusCode,
  }: GetResponseAdapterProps<ALBEvent>): ALBResult {
    const multiValueHeaders = !event.headers
      ? getMultiValueHeadersMap(responseHeaders)
      : undefined;

    const headers = event.headers
      ? getFlattenedHeadersMap(responseHeaders)
      : undefined;
   
    // credits to: https://github.com/dougmoscrop/serverless-http/commit/e41bd784e3351d18b7c3f8aec4fcead764f396af
    if ((headers && headers['transfer-encoding'] === 'chunked') || (multiValueHeaders && multiValueHeaders['transfer-encoding']?.includes('chunked')) || response.chunkedEncoding) {
        const raw = body.split('\r\n');
        const parsed = [];
        for (let i = 0; i < raw.length; i +=2) {
          const size = parseInt(raw[i], 16);
          const value = raw[i + 1];
          if (value) {
            parsed.push(value.substring(0, size));
          }
        }
        body = parsed.join()
      }

    return {
      statusCode,
      body,
      headers,
      multiValueHeaders,
      isBase64Encoded,
    };
  }
```

To modify this piece of code, you can basically go to `alb.adapter.ts`, copy the entire file to your project, and then do the modification of the `body` inside `getResponse`.

This will probably fix your issue with `Transfer-Encoding: Chunked`.

If it fixes your issue, please open a PR!

Testing ALB for me is a little bit difficult since it takes a lot of effort to just create the whole environment, so if you can contribute/test, I will be very happy.

**Comment by H4ad** [comment]
The #167 should fix this issue, I should merge and release a new version tomorrow.

**Comment by H4ad** [comment]
Released as [4.1.0](https://www.npmjs.com/package/@h4ad/serverless-adapter/v/4.1.0)

Thanks for opening this issue, let me know if you found any other issue.
